test/extended/util/client.go

```diff
@@ -390,6 +393,21 @@ func (c *CLI) TeardownProject() {
 		err := dynamicClient.Resource(resource.Resource).Namespace(resource.Namespace).Delete(context.Background(), resource.Name, metav1.DeleteOptions{})
 		framework.Logf("Deleted %v, err: %v", resource, err)
 	}
+
+	// Point back at the original config file as this CLI
+	// object can continue to be used in other Teardown blocks
+	if c.origAdminConfigPath != "" {
+		os.Remove(c.adminConfigPath)
+		c.adminConfigPath = c.origAdminConfigPath
+		c.origAdminConfigPath = ""
+	}
+}
+
+// SetupCLI takes a copy of KUBECONFIG to ensure the cli doesn't write to the original
+// and doesn't read the origional as often (reducing the likelyhood of being read while being written to by another test)
+func (c *CLI) SetupCLI() {
+	c.origAdminConfigPath = c.adminConfigPath
+	c.adminConfigPath = copyToTmp(c.adminConfigPath)
```

Comment:
Are we sure this is safe? In the parallel suite, are multiple tests using the same CLI?

Reply:
I believe its safe, I've checked it with some tests in parallel and each one had its own CLI object. 
Also other methods in CLI  update c.configPath, if it was a problem the tests would have shown this up by now.